src/qt/walletmodel.cpp

```diff
@@ -878,6 +878,13 @@ WalletModel::SendCoinsReturn WalletModel::prepareSigmaSpendTransaction(
         *newTx = wallet->CreateSigmaSpendTransaction(sendRecipients, fee, selectedCoins, changes, coinControl);
     } catch (const InsufficientFunds& err) {
         return AmountExceedsBalance;
+    } catch (const std::runtime_error& err) {
+        if (0 == std::strcmp(
+            err.what(),
+            std::runtime_error(_("Can not choose coins within limit.")).what())) {
+            return ExceedLimit;
+        }
+        throw err;
```

Comment:
why simple text comparison is so weird here?

Reply:
I mean why this
``if (`_("error message") == err.what())```
won't work?

Reply:
It don't work because return value of `what` is a pointer to a null-terminated string.

https://en.cppreference.com/w/cpp/error/exception/what

Reply:
thanks for fix. I don't know std::string and null-terminated string can compare this way before.

https://en.cppreference.com/w/cpp/string/basic_string/operator_cmp